<center>
    <img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DA0101EN-SkillsNetwork/labs/Module%201/images/IDSNlogo.png" width="300" alt="cognitiveclass.ai logo"  />
</center>

# Introduction  Notebook

Estimated time needed: **10** minutes

## Objectives

After completing this lab you will be able to:

*   Acquire data in various ways
*   Obtain insights from data with Pandas library


<h2>Table of Contents</h2>

<div class="alert alert-block alert-info" style="margin-top: 20px">
<ol>
    <li><a href="https://#data_acquisition">Data Acquisition</a>
    <li><a href="https://#data_acquisition">Data Acquisition</a>
    <li><a href="https://#basic_insight">Basic Insight of Dataset</a></li>
</ol>

</div>
<hr>


<h1 id="data_acquisition">Data Acquisition</h1>
<p>
There are various formats for a dataset: .csv, .json, .xlsx  etc. The dataset can be stored in different places, on your local machine or sometimes online.<br>

In this section, you will learn how to load a dataset into our Jupyter Notebook.<br>

In our case, the Diabetes Dataset is an online source, and it is in a CSV (comma separated value) format. Let's use this dataset as an example to practice data reading.

<ul>
    <li>Data source: <a href="https://www.kaggle.com/datasets/brandao/diabetes" target="_blank">https://www.kaggle.com/datasets/brandao/diabetes</a></li>
    <li>Data type: csv</li>
</ul>

The statistical data obtained from https://www.kaggle.com/datasets/brandao/diabetes under [CC0: Public Domain](https://creativecommons.org/publicdomain/zero/1.0/) license.

The Pandas Library is a useful tool that enables us to read various datasets into a dataframe; our Jupyter notebook platforms have a built-in <b>Pandas Library</b> so that all we need to do is import Pandas without installing.
</p>


you are running the lab in your  browser, so we will install the libraries using `piplite`


In [349]:
#you are running the lab in your  browser, so we will install the libraries using ``piplite``
#import piplite
#import micropip
#await piplite.install(['pandas'])
#await piplite.install(['matplotlib'])
#await piplite.install(['scipy'])
#await piplite.install(['seaborn'])
#await micropip.install(['ipywidgets'],keep_going=True)
#await micropip.install(['tqdm'],keep_going=True)


If you run the lab locally using Anaconda, you can load the correct library and versions by uncommenting the following:


In [350]:
#install specific version of libraries used in  lab
#! mamba install pandas==1.3.3  -y
#! mamba install numpy=1.21.2 -y

In [351]:
# import pandas library
import pandas as pd
import numpy as np

This function will download the dataset into your browser


In [352]:
#This function will download the dataset into your browser 

#from pyodide.http import pyfetch
#
#async def download(url, filename):
#    response = await pyfetch(url)
#    if response.status == 200:
#        with open(filename, "wb") as f:
#            f.write(await response.bytes())

<h2>Read Data</h2>
<p>
We use <code>pandas.read_csv()</code> function to read the csv file. In the brackets, we put the file path along with a quotation mark so that pandas will read the file into a dataframe from that address. The file path can be either an URL or your local file address.<br>

You can also assign the dataset to any variable you create.

</p>


In [353]:
path = "diabetic_data.csv"

you will need to download the dataset; if you are running locally, please comment out the following


In [354]:
#you will need to download the dataset; if you are running locally, please comment out the following 
# await download(path, "diabetic_data.csv")
# path="diabetic_data.csv"

In [355]:
# Import pandas library
import pandas as pd

# Read the online file by the URL provides above, and assign it to variable "df"

df = pd.read_csv(path)

After reading the dataset, we can use the <code>dataframe.head(n)</code> method to check the top n rows of the dataframe, where n is an integer. Contrary to <code>dataframe.head(n)</code>, <code>dataframe.tail(n)</code> will show you the bottom n rows of the dataframe.


In [356]:
# show the first 5 rows using dataframe.head() method
print("The first 5 rows of the dataframe") 
df.head(5)

The first 5 rows of the dataframe


encounter_id  patient_nbr             race  gender      age weight  \
0       2278392      8222157        Caucasian  Female   [0-10)      ?   
1        149190     55629189        Caucasian  Female  [10-20)      ?   
2         64410     86047875  AfricanAmerican  Female  [20-30)      ?   
3        500364     82442376        Caucasian    Male  [30-40)      ?   
4         16680     42519267        Caucasian    Male  [40-50)      ?   

   admission_type_id  discharge_disposition_id  admission_source_id  \
0                  6                        25                    1   
1                  1                         1                    7   
2                  1                         1                    7   
3                  1                         1                    7   
4                  1                         1                    7   

   time_in_hospital  ... citoglipton insulin  glyburide-metformin  \
0                 1  ...          No      No                   No   
1                 3  ...          No      Up                   No   
2                 2  ...          No      No                   No   
3                 2  ...          No      Up                   No   
4                 1  ...          No  Steady                   No   

   glipizide-metformin  glimepiride-pioglitazone  metformin-rosiglitazone  \
0                   No                        No                       No   
1                   No                        No                       No   
2                   No                        No                       No   
3                   No                        No                       No   
4                   No                        No                       No   

   metformin-pioglitazone  change diabetesMed readmitted  
0                      No      No          No         NO  
1                      No      Ch         Yes        >30  
2                      No      No         Yes         NO  
3                      No      Ch         Yes         NO  
4                      No      Ch         Yes         NO  

[5 rows x 50 columns]

<div class="alert alert-danger alertdanger" style="margin-top: 20px">
<h1> Question #1: </h1>
<b>Check the bottom 10 rows of data frame "df".</b>
</div>


In [357]:
# Write your code below and press Shift+Enter to execute 


<details><summary>Click here for the solution</summary>

```python
    print("The last 10 rows of the dataframe\n")
    df.tail(10)
```

<h3>Add Headers</h3>
<p>
Take a look at our dataset. Pandas automatically set the header with an integer starting from 0.
</p>
<p>
To better describe our data, we can introduce a header. This information is available at:  <a href="https://www.kaggle.com/datasets/brandao/diabetes" target="_blank">https://www.kaggle.com/datasets/brandao/diabetes</a>.
</p>
<p>
To make our header look better, we can override it.
</p>
<p>
First, we create a list "headers" that include all column names in order.
Then, we use <code>dataframe.columns = headers</code> to replace the headers with the list we created.
</p>

In [358]:
# create headers list
headers = ["EncounterId","PatientNbr","Race","Gender","Age","Weight","AdmissionTypeId",
           "DischargeDispositionId","AdmissionSourceId","TimeInHospital","PayerCode",
           "MedicalSpecialty","NumLabProcedures","NumProcedures","NumMedications",
           "NumberOutpatient","NumberEmergency","NumberInpatient","Diagnosis1","Diagnosis2",
           "Diagnosis3","NumberDiagnoses","MaxGluSerum","A1Cresult","Metformin","Repaglinide",
           "Nateglinide","Chlorpropamide","Glimepiride","Acetohexamide","Glipizide",
           "Glyburide","Tolbutamide","Pioglitazone","Rosiglitazone","Acarbose","Miglitol",
           "Troglitazone","Tolazamide","Examide","Citoglipton","Insulin","Glyburide-metformin",
           "Glipizide-metformin","Glimepiride-pioglitazone","Metformin-rosiglitazone","Metformin-pioglitazone",
           "Change","DiabetesMed","Readmitted"]
print("headers\n", headers)

headers
 ['EncounterId', 'PatientNbr', 'Race', 'Gender', 'Age', 'Weight', 'AdmissionTypeId', 'DischargeDispositionId', 'AdmissionSourceId', 'TimeInHospital', 'PayerCode', 'MedicalSpecialty', 'NumLabProcedures', 'NumProcedures', 'NumMedications', 'NumberOutpatient', 'NumberEmergency', 'NumberInpatient', 'Diagnosis1', 'Diagnosis2', 'Diagnosis3', 'NumberDiagnoses', 'MaxGluSerum', 'A1Cresult', 'Metformin', 'Repaglinide', 'Nateglinide', 'Chlorpropamide', 'Glimepiride', 'Acetohexamide', 'Glipizide', 'Glyburide', 'Tolbutamide', 'Pioglitazone', 'Rosiglitazone', 'Acarbose', 'Miglitol', 'Troglitazone', 'Tolazamide', 'Examide', 'Citoglipton', 'Insulin', 'Glyburide-metformin', 'Glipizide-metformin', 'Glimepiride-pioglitazone', 'Metformin-rosiglitazone', 'Metformin-pioglitazone', 'Change', 'DiabetesMed', 'Readmitted']


In [359]:
df.columns = headers
df.head(10)

EncounterId  PatientNbr             Race  Gender       Age Weight  \
0      2278392     8222157        Caucasian  Female    [0-10)      ?   
1       149190    55629189        Caucasian  Female   [10-20)      ?   
2        64410    86047875  AfricanAmerican  Female   [20-30)      ?   
3       500364    82442376        Caucasian    Male   [30-40)      ?   
4        16680    42519267        Caucasian    Male   [40-50)      ?   
5        35754    82637451        Caucasian    Male   [50-60)      ?   
6        55842    84259809        Caucasian    Male   [60-70)      ?   
7        63768   114882984        Caucasian    Male   [70-80)      ?   
8        12522    48330783        Caucasian  Female   [80-90)      ?   
9        15738    63555939        Caucasian  Female  [90-100)      ?   

   AdmissionTypeId  DischargeDispositionId  AdmissionSourceId  TimeInHospital  \
0                6                      25                  1               1   
1                1                       1                  7               3   
2                1                       1                  7               2   
3                1                       1                  7               2   
4                1                       1                  7               1   
5                2                       1                  2               3   
6                3                       1                  2               4   
7                1                       1                  7               5   
8                2                       1                  4              13   
9                3                       3                  4              12   

   ... Citoglipton Insulin  Glyburide-metformin  Glipizide-metformin  \
0  ...          No      No                   No                   No   
1  ...          No      Up                   No                   No   
2  ...          No      No                   No                   No   
3  ...          No      Up                   No                   No   
4  ...          No  Steady                   No                   No   
5  ...          No  Steady                   No                   No   
6  ...          No  Steady                   No                   No   
7  ...          No      No                   No                   No   
8  ...          No  Steady                   No                   No   
9  ...          No  Steady                   No                   No   

   Glimepiride-pioglitazone  Metformin-rosiglitazone  Metformin-pioglitazone  \
0                        No                       No                      No   
1                        No                       No                      No   
2                        No                       No                      No   
3                        No                       No                      No   
4                        No                       No                      No   
5                        No                       No                      No   
6                        No                       No                      No   
7                        No                       No                      No   
8                        No                       No                      No   
9                        No                       No                      No   

   Change DiabetesMed Readmitted  
0      No          No         NO  
1      Ch         Yes        >30  
2      No         Yes         NO  
3      Ch         Yes         NO  
4      Ch         Yes         NO  
5      No         Yes        >30  
6      Ch         Yes         NO  
7      No         Yes        >30  
8      Ch         Yes         NO  
9      Ch         Yes         NO  

[10 rows x 50 columns]

We need to replace the "?" symbol with NaN so the dropna() can remove the missing values:


In [360]:
df1=df.replace('?',np.NaN)


We can drop missing values along the column "weight" as follows:


In [361]:
df=df1.dropna(subset=["Weight"], axis=0)
df.head(20)

EncounterId  PatientNbr             Race  Gender      Age     Weight  \
1102      8042082    94466574        Caucasian    Male  [50-60)   [75-100)   
1108      8057670    96198543              NaN  Female  [80-90)    [50-75)   
1132      8235036    79874631        Caucasian  Female  [60-70)     [0-25)   
1136      8274126    90421380        Caucasian  Female  [40-50)   [75-100)   
1190      8421486    55628172        Caucasian  Female  [70-80)   [75-100)   
1204      8576406    80041266        Caucasian  Female  [80-90)    [50-75)   
1214      8620524    93126573        Caucasian  Female  [50-60)   [75-100)   
1230      8753532    10648701        Caucasian    Male  [70-80)   [75-100)   
1256      8866632   103586670        Caucasian    Male  [70-80)  [100-125)   
1294      9075132    96007464  AfricanAmerican  Female  [60-70)   [75-100)   
1295      9076452    67852377              NaN    Male  [50-60)   [75-100)   
1298      9089646    42912702        Caucasian  Female  [60-70)    [50-75)   
1313      9198750    81133947        Caucasian    Male  [50-60)   [75-100)   
1332      9327402    78871527  AfricanAmerican    Male  [40-50)   [75-100)   
1433      9813246    42516765        Caucasian    Male  [70-80)   [75-100)   
1515     10320138    50444955        Caucasian    Male  [50-60)   [75-100)   
1678     11190414    25739460        Caucasian    Male  [60-70)   [75-100)   
1683     11194338   100526274        Caucasian    Male  [70-80)   [75-100)   
2465     16173522    54126873        Caucasian  Female  [70-80)    [50-75)   
2485     16203084   105841233        Caucasian    Male  [70-80)   [75-100)   

      AdmissionTypeId  DischargeDispositionId  AdmissionSourceId  \
1102                6                       1                 17   
1108                6                       5                 17   
1132                6                       1                 17   
1136                6                       1                 17   
1190                6                       6                 17   
1204                6                       1                 17   
1214                6                       1                 17   
1230                6                       3                 17   
1256                6                       1                 17   
1294                6                       3                 17   
1295                6                       6                 17   
1298                6                       1                 17   
1313                6                       1                 17   
1332                6                       1                 17   
1433                6                       6                 17   
1515                6                       1                 17   
1678                6                       1                 17   
1683                6                       6                 17   
2465                6                       6                 17   
2485                6                       1                 17   

      TimeInHospital  ... Citoglipton Insulin  Glyburide-metformin  \
1102               2  ...          No  Steady                   No   
1108               9  ...          No  Steady                   No   
1132               5  ...          No  Steady                   No   
1136               3  ...          No  Steady                   No   
1190              11  ...          No  Steady                   No   
1204               9  ...          No  Steady                   No   
1214               6  ...          No      No                   No   
1230               6  ...          No      No                   No   
1256               6  ...          No  Steady                   No   
1294               9  ...          No  Steady                   No   
1295               6  ...          No  Steady                   No   
1298               7  ...          No  Steady                   No   
1313               2  ...   

Now, we have successfully read the raw dataset and added the correct headers into the dataframe.


 <div class="alert alert-danger alertdanger" style="margin-top: 20px">
<h1> Question #2: </h1>
<b>Find the name of the columns of the dataframe.</b>
</div>


In [362]:
# Write your code below and press Shift+Enter to execute


<details><summary>Click here for the solution</summary>

```python
    print(df.columns)
```
</details>


<h2>Save Dataset</h2>
<p>
Correspondingly, Pandas enables us to save the dataset to csv. By using the <code>dataframe.to_csv()</code> method, you can add the file path and name along with quotation marks in the brackets.
</p>
<p>
For example, if you would save the dataframe <b>df</b> as <b>diabetes.csv</b> to your local machine, you may use the syntax below, where <code>index = False</code> means the row names will not be written.
</p>


We can also read and save other file formats. We can use similar functions like **`pd.read_csv()`** and **`df.to_csv()`** for other data formats. The functions are listed in the following table:


<h2>Read/Save Other Data Formats</h2>

| Data Formate |        Read       |            Save |
| ------------ | :---------------: | --------------: |
| csv          |  `pd.read_csv()`  |   `df.to_csv()` |
| json         |  `pd.read_json()` |  `df.to_json()` |
| excel        | `pd.read_excel()` | `df.to_excel()` |
| hdf          |  `pd.read_hdf()`  |   `df.to_hdf()` |
| sql          |  `pd.read_sql()`  |   `df.to_sql()` |
| ...          |        ...        |             ... |


<h1 id="basic_insight">Basic Insight of Dataset</h1>
<p>
After reading data into Pandas dataframe, it is time for us to explore the dataset.<br>

There are several ways to obtain essential insights of the data to help us better understand our dataset.

</p>


<h2>Data Types</h2>
<p>
Data has a variety of types.<br>

The main types stored in Pandas dataframes are <b>object</b>, <b>float</b>, <b>int</b>, <b>bool</b> and <b>datetime64</b>. In order to better learn about each attribute, it is always good for us to know the data type of each column. In Pandas:

</p>


In [363]:
df.dtypes


EncounterId                  int64
PatientNbr                   int64
Race                        object
Gender                      object
Age                         object
Weight                      object
AdmissionTypeId              int64
DischargeDispositionId       int64
AdmissionSourceId            int64
TimeInHospital               int64
PayerCode                   object
MedicalSpecialty            object
NumLabProcedures             int64
NumProcedures                int64
NumMedications               int64
NumberOutpatient             int64
NumberEmergency              int64
NumberInpatient              int64
Diagnosis1                  object
Diagnosis2                  object
Diagnosis3                  object
NumberDiagnoses              int64
MaxGluSerum                 object
A1Cresult                   object
Metformin                   object
Repaglinide                 object
Nateglinide                 object
Chlorpropamide              object
Glimepiride         

A series with the data type of each column is returned.


In [364]:
# check the data type of data frame "df" by .dtypes
print(df.dtypes)

EncounterId                  int64
PatientNbr                   int64
Race                        object
Gender                      object
Age                         object
Weight                      object
AdmissionTypeId              int64
DischargeDispositionId       int64
AdmissionSourceId            int64
TimeInHospital               int64
PayerCode                   object
MedicalSpecialty            object
NumLabProcedures             int64
NumProcedures                int64
NumMedications               int64
NumberOutpatient             int64
NumberEmergency              int64
NumberInpatient              int64
Diagnosis1                  object
Diagnosis2                  object
Diagnosis3                  object
NumberDiagnoses              int64
MaxGluSerum                 object
A1Cresult                   object
Metformin                   object
Repaglinide                 object
Nateglinide                 object
Chlorpropamide              object
Glimepiride         

<p>
As shown above, it is clear to see that the data type of "EncounterId" and "PatientNbr" are <code>int64</code>, "Race" is <code>object</code>, etc.
</p>
<p>
These data types can be changed; we will learn how to accomplish this in a later module.
</p>


<h2>Describe</h2>
If we would like to get a statistical summary of each column e.g. count, column mean value, column standard deviation, etc., we use the describe method:


This method will provide various summary statistics, excluding <code>NaN</code> (Not a Number) values.


In [365]:
df.describe()

EncounterId    PatientNbr  AdmissionTypeId  DischargeDispositionId  \
count  3.197000e+03  3.197000e+03      3197.000000             3197.000000   
mean   1.559442e+08  7.121929e+07         2.388176                2.712230   
std    1.010020e+08  3.112114e+07         1.748526                2.850754   
min    8.042082e+06  1.629000e+03         1.000000                1.000000   
25%    8.323027e+07  5.422239e+07         1.000000                1.000000   
50%    1.277816e+08  7.670522e+07         2.000000                1.000000   
75%    2.127792e+08  9.572209e+07         3.000000                3.000000   
max    4.436958e+08  1.867680e+08         6.000000               18.000000   

       AdmissionSourceId  TimeInHospital  NumLabProcedures  NumProcedures  \
count        3197.000000     3197.000000       3197.000000    3197.000000   
mean            6.112293        4.803253         53.142321       1.491711   
std             4.962522        2.991919         18.970172       1.783584   
min             1.000000        1.000000          1.000000       0.000000   
25%             1.000000        3.000000         41.000000       0.000000   
50%             7.000000        4.000000         55.000000       1.000000   
75%             7.000000        6.000000         68.000000       2.000000   
max            17.000000       14.000000        105.000000       6.000000   

       NumMedications  NumberOutpatient  NumberEmergency  NumberInpatient  \
count     3197.000000       3197.000000      3197.000000      3197.000000   
mean        16.351580          1.089459         0.233969         0.574601   
std          7.885302          1.874539         0.758141         1.219299   
min          1.000000          0.000000         0.000000         0.000000   
25%         11.000000          0.000000         0.000000         0.000000   
50%         15.000000          0.000000         0.000000         0.000000   
75%         20.000000          2.000000         0.000000         1.000000   
max         65.000000         21.000000        13.000000        21.000000   

       NumberDiagnoses  
count      3197.000000  
mean          7.975915  
std           1.526276  
min           1.000000  
25%           7.000000  
50%           9.000000  
75%           9.000000  
max           9.000000

<p>
This shows the statistical summary of all numeric-typed (int, float) columns.<br>

For example, the attribute "DischargeDispositionId" has 3197 counts, the mean value of this column is 2.71, the standard deviation is 2.85, the minimum value is 1, 25th percentile is 1, 50th percentile is 1, 75th percentile is 3, and the maximum value is 18. <br>

However, what if we would also like to check all the columns including those that are of type object? <br><br>

You can add an argument <code>include = "all"</code> inside the bracket. Let's try it again.

</p>


In [366]:
# describe all the columns in "df" 
df.describe(include = "all")

EncounterId    PatientNbr       Race  Gender      Age    Weight  \
count   3.197000e+03  3.197000e+03       3059    3197     3197      3197   
unique           NaN           NaN          4       2       10         9   
top              NaN           NaN  Caucasian  Female  [70-80)  [75-100)   
freq             NaN           NaN       2907    1648      938      1336   
mean    1.559442e+08  7.121929e+07        NaN     NaN      NaN       NaN   
std     1.010020e+08  3.112114e+07        NaN     NaN      NaN       NaN   
min     8.042082e+06  1.629000e+03        NaN     NaN      NaN       NaN   
25%     8.323027e+07  5.422239e+07        NaN     NaN      NaN       NaN   
50%     1.277816e+08  7.670522e+07        NaN     NaN      NaN       NaN   
75%     2.127792e+08  9.572209e+07        NaN     NaN      NaN       NaN   
max     4.436958e+08  1.867680e+08        NaN     NaN      NaN       NaN   

        AdmissionTypeId  DischargeDispositionId  AdmissionSourceId  \
count       3197.000000             3197.000000        3197.000000   
unique              NaN                     NaN                NaN   
top                 NaN                     NaN                NaN   
freq                NaN                     NaN                NaN   
mean           2.388176                2.712230           6.112293   
std            1.748526                2.850754           4.962522   
min            1.000000                1.000000           1.000000   
25%            1.000000                1.000000           1.000000   
50%            2.000000                1.000000           7.000000   
75%            3.000000                3.000000           7.000000   
max            6.000000               18.000000          17.000000   

        TimeInHospital  ... Citoglipton Insulin  Glyburide-metformin  \
count      3197.000000  ...        3197    3197                 3197   
unique             NaN  ...           1       4                    1   
top                NaN  ...          No      No                   No   
freq               NaN  ...        3197    2288                 3197   
mean          4.803253  ...         NaN     NaN                  NaN   
std           2.991919  ...         NaN     NaN                  NaN   
min           1.000000  ...         NaN     NaN                  NaN   
25%           3.000000  ...         NaN     NaN                  NaN   
50%           4.000000  ...         NaN     NaN                  NaN   
75%           6.000000  ...         NaN     NaN                  NaN   
max          14.000000  ...         NaN     NaN                  NaN   

        Glipizide-metformin  Glimepiride-pioglitazone  \
count                  3197                      3197   
unique                    1                         1   
top                      No                        No   
freq                   3197                      3197   
mean                    NaN                       NaN   
std                     NaN                       NaN   
min                     NaN                       NaN   
25%                     NaN                       NaN   
50%                     NaN                       NaN   
75%                     NaN                       NaN   
max                     NaN                       NaN   

        Metformin-rosiglitazone  Metformin-pioglitazone  Change DiabetesMed  \
count                      3197                    3197    3197        3197   
unique                        1                       1       2           2   
top                          No                      No      No         Yes   
freq                       3197                    3197    2131        2201   
mean                        NaN                     NaN     NaN         NaN   
std                         NaN                     NaN     NaN         NaN   
min                         NaN                     NaN     NaN         NaN   
25%                         NaN                     NaN     NaN         N

<p>
Now it provides the statistical summary of all the columns, including object-typed attributes.<br>

We can now see how many unique values there, which one is the top value and the frequency of top value in the object-typed columns.<br>

Some values in the table above show as "NaN". This is because those numbers are not available regarding a particular column type.<br>

</p>

<div class="alert alert-danger alertdanger" style="margin-top: 20px">
<h1> Question #3: </h1>

<p>
You can select the columns of a dataframe by indicating the name of each column. For example, you can select the three columns as follows:
</p>
<p>
    <code>dataframe[[' column 1 ',column 2', 'column 3']]</code>
</p>
<p>
Where "column" is the name of the column, you can apply the method  ".describe()" to get the statistics of those columns as follows:
</p>
<p>
    <code>dataframe[[' column 1 ',column 2', 'column 3'] ].describe()</code>
</p>

Apply the  method to ".describe()" to the columns 'Age' and 'Weight'.

</div>


In [367]:
# Write your code below and press Shift+Enter to execute

<details><summary>Click here for the solution</summary>

```python
    df[['Age', 'Weight']].describe()
```
</details>

<h2>Info</h2>
Another method you can use to check your dataset is:


It provides a concise summary of your DataFrame.

This method prints information about a DataFrame including the index dtype and columns, non-null values and memory usage.


In [368]:
# look at the info of "df"
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3197 entries, 1102 to 101723
Data columns (total 50 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   EncounterId               3197 non-null   int64 
 1   PatientNbr                3197 non-null   int64 
 2   Race                      3059 non-null   object
 3   Gender                    3197 non-null   object
 4   Age                       3197 non-null   object
 5   Weight                    3197 non-null   object
 6   AdmissionTypeId           3197 non-null   int64 
 7   DischargeDispositionId    3197 non-null   int64 
 8   AdmissionSourceId         3197 non-null   int64 
 9   TimeInHospital            3197 non-null   int64 
 10  PayerCode                 2315 non-null   object
 11  MedicalSpecialty          1646 non-null   object
 12  NumLabProcedures          3197 non-null   int64 
 13  NumProcedures             3197 non-null   int64 
 14  NumMedications     

<h1>Excellent! You have just completed the  Introduction Notebook!</h1>


### Thank you for completing this lab!

## Author

<a href="https://www.linkedin.com/in/joseph-s-50398b136/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDA0101ENSkillsNetwork20235326-2021-01-01" target="_blank">Joseph Santarcangelo</a>

### Other Contributors

<a href="https://www.linkedin.com/in/mahdi-noorian-58219234/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDA0101ENSkillsNetwork20235326-2021-01-01" target="_blank">Mahdi Noorian PhD</a>

Bahare Talayian

Eric Xiao

Steven Dong

Parizad

Hima Vasudevan

<a href="https://www.linkedin.com/in/fiorellawever/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDA0101ENSkillsNetwork20235326-2021-01-01" target="_blank">Fiorella Wenver</a>

<a href="https:// https://www.linkedin.com/in/yi-leng-yao-84451275/ " target="_blank" >Yi Yao</a>.

## Change Log

| Date (YYYY-MM-DD) | Version | Changed By | Change Description                                         |
| ----------------- | ------- | ---------- | ---------------------------------------------------------- |
| 2022-08-23        | 2.4     | Malika     | Import micropip added and parameter for ipwidgets and tqdm |
| 2020-10-30        | 2.3     | Lakshmi    | Changed URL of the csv                                     |
| 2020-09-22        | 2.2     | Nayef      | Added replace() method to remove '?'                       |
| 2020-09-09        | 2.1     | Lakshmi    | Made changes in info method of dataframe                   |
| 2020-08-27        | 2.0     | Lavanya    | Moved lab to course repo in GitLab                         |

<hr>

## <h3 align="center"> © IBM Corporation 2020. All rights reserved. <h3/>
